src/accordion/AccordionState.ts

```diff
@@ -147,7 +187,55 @@ function reducer(
         return { ...state, items: [item] };
       }
 
-      return { ...state, items: [...state.items, item] };
+      return { ...state, items: [...items, item] };
+    }
+
+    case "next": {
+      const { id } = action;
+
+      const currentIndex = buttonIds.indexOf(id);
+      const nextIndex = (currentIndex + 1) % total;
+
+      if (!loop && nextIndex === 0) {
+        return { ...state };
+      }
+
+      const nextItem = buttons[nextIndex];
+      nextItem.ref?.current?.focus();
+
+      return { ...state };
+    }
+
+    case "prev": {
+      const { id } = action;
+
+      const currentIndex = buttonIds.indexOf(id);
+      const prevIndex = (currentIndex - 1 + total) % total;
+
+      if (!loop && prevIndex === total - 1) {
+        return { ...state };
+      }
+
+      const prevItem = buttons[prevIndex];
+      prevItem.ref?.current?.focus();
+
+      return { ...state };
+    }
+
+    case "first": {
+      const firstItem = buttons[0];
+      firstItem.ref?.current?.focus();
+
+      return { ...state };
+    }
+
+    case "last": {
+      const total = buttons.length;
+
+      const lastItem = buttons[total - 1];
+      lastItem.ref?.current?.focus();
+
+      return { ...state };
```

Comment:
Seems like we can also utilize a helper function here to remove the redundancy.

Reply:
Yup, I thought that too.

I just procrastinated it yesterday 😛 because of burn out(not in the mood)
I will do it today.

Reply:
Okay no worries, not high priority you can also do this on a different PR.

Reply:
I have made some refactoring by removing the logic from  reducer and add it to the hook directly.

Let me know, if any more can be done.

Reply:
Yeah looks nice!